webapp/src/test/java/com/box/l10n/mojito/android/strings/XmlWriterTest.java

```diff
@@ -0,0 +1,164 @@
+package com.box.l10n.mojito.android.strings;
+
+import org.junit.Test;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.box.l10n.mojito.android.strings.XmlHelper.*;
+import static org.junit.Assert.*;
+
+public class XmlWriterTest {
+
+    @Test
+    public void addStringTest() throws TransformerException, ParserConfigurationException {
+        XmlDocument xmlDocument = new XmlDocument()
+                .addSingle("comment", "name", "content");
+        
+        assertEquals(
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?><resources>\n" +
+                        "<!--comment-->\n" +
+                        "<string name=\"name\">content</string>\n" +
+                        "</resources>\n",
+                new XmlWriter().toText(xmlDocument.getItems()));
+    }
+
+    @Test
+    public void addQuotesTest() throws TransformerException, ParserConfigurationException {
+        XmlDocument xmlDocument = new XmlDocument()
```

Comment:
Why do we need this XmlDocument abstraction? Why not write just as they come without this intermediate data structure. 

As a end user, I just need a very simple API:

``` 
asw = new AndroidStringWritter();
asw.addString();
asw.addPluralString();
asw.toXml(); // or asw.toString();
```

Or if you using extra object at least a more explicit naming convention, something like:

```
textUnits = new TextUnits();
textUnits.addSingular("name", "content", "comment");
textUnits.addPlural(...);

asw = new AndroidStringWritter();
asw.toText(textUnits);
```